ledger/participant-integration-api/src/main/scala/platform/apiserver/ApiServerConfig.scala

```diff
@@ -4,31 +4,62 @@
 package com.daml.platform.apiserver
 
 import com.daml.ledger.api.tls.TlsConfiguration
-import com.daml.lf.data.Ref
 import com.daml.platform.apiserver.SeedService.Seeding
-import com.daml.platform.configuration.{IndexConfiguration, InitialLedgerConfiguration}
-import com.daml.ports.Port
-import java.nio.file.Path
-import java.time.Duration
-
+import com.daml.platform.configuration.{
+  CommandConfiguration,
+  InitialLedgerConfiguration,
+  PartyConfiguration,
+}
+import com.daml.platform.services.time.TimeProviderType
+import com.daml.platform.store.DbSupport.{ConnectionPoolConfig, DbConfig}
 import com.daml.platform.usermanagement.UserManagementConfig
+import com.daml.ports.Port
 
-import scala.concurrent.duration.FiniteDuration
+import java.nio.file.Path
+import scala.concurrent.duration._
 
 case class ApiServerConfig(
-    participantId: Ref.ParticipantId,
-    port: Port,
-    address: Option[String], // This defaults to "localhost" when set to `None`.
-    jdbcUrl: String,
-    databaseConnectionPoolSize: Int,
-    databaseConnectionTimeout: FiniteDuration,
-    tlsConfig: Option[TlsConfiguration],
-    maxInboundMessageSize: Int,
-    initialLedgerConfiguration: Option[InitialLedgerConfiguration],
-    configurationLoadTimeout: Duration,
-    indexConfiguration: IndexConfiguration,
-    portFile: Option[Path],
-    seeding: Seeding,
-    managementServiceTimeout: Duration,
-    userManagementConfig: UserManagementConfig,
+    address: Option[String] =
+      ApiServerConfig.DefaultAddress, // This defaults to "localhost" when set to `None`.
+    apiStreamShutdownTimeout: Duration = ApiServerConfig.DefaultApiStreamShutdownTimeout,
+    authentication: AuthServiceConfig = ApiServerConfig.DefaultAuthentication,
+    command: CommandConfiguration = ApiServerConfig.DefaultCommand,
+    configurationLoadTimeout: Duration = ApiServerConfig.DefaultConfigurationLoadTimeout,
+    database: DbConfig = ApiServerConfig.DefaultDatabase,
```

Comment:
I think this parameter better fits in `IndexServiceConfig`

Reply:
`database` which is initialised from this configuration is then shared with index service and user management. I would need to duplicate this parameter for both then. wdyt?

Reply:
Why do we need to duplicate? Can't we pass it to wherever we need it as `participantConfig.indexService.database`? I am not adamant on this, but I observed that `database` is the only field from `ApiServerConfig` not actually used in `StandaloneApiServer`.